component/src/main/java/io/siddhi/extension/execution/file/FileCopyExtension.java

```diff
@@ -0,0 +1,294 @@
+            }
+            Attribute.Type firstAttributeType = attributeExpressionExecutors[0].getReturnType();
+            if (!(firstAttributeType == Attribute.Type.STRING)) {
+                throw new SiddhiAppValidationException("Invalid parameter type found for uri " +
+                        "(first argument) of file:copy() function, required " + Attribute.Type.STRING +
+                        " but found " + firstAttributeType.toString());
+            }
+            if (attributeExpressionExecutors[1] == null) {
+                throw new SiddhiAppValidationException("Invalid input given to destination.dir.uri (second argument) " +
+                        "file:copy() function. Argument cannot be null");
+            }
+            Attribute.Type secondAttributeType = attributeExpressionExecutors[1].getReturnType();
+            if (!(secondAttributeType == Attribute.Type.STRING)) {
+                throw new SiddhiAppValidationException("Invalid parameter type found for destination.dir.uri " +
+                        "(second argument) of file:copy() function, required " + Attribute.Type.STRING +
+                        " but found " + firstAttributeType.toString());
+            }
+            if (attributeExpressionExecutors.length == 3) {
+                Attribute.Type thirdAttributeType = attributeExpressionExecutors[2].getReturnType();
+                if (!(thirdAttributeType == Attribute.Type.STRING)) {
+                    throw new SiddhiAppValidationException("Invalid parameter type found for regex " +
+                            "(third argument) of file:copy() function, required " + Attribute.Type.STRING +
+                            " but found " + firstAttributeType.toString());
+                }
+            }
+            if (attributeExpressionExecutors.length == 4) {
+                Attribute.Type thirdAttributeType = attributeExpressionExecutors[3].getReturnType();
+                if (!(thirdAttributeType == Attribute.Type.BOOL)) {
+                    throw new SiddhiAppValidationException("Invalid parameter type found for exclude.root.folder " +
+                            "(forth argument) of file:copy() function, required " + Attribute.Type.BOOL +
+                            " but found " + firstAttributeType.toString());
+                }
+            }
+        } else {
+            throw new SiddhiAppValidationException("Invalid no of arguments passed to file:copy() function, "
+                    + "required 2, 3 or 4, but found " + attributeExpressionExecutors.length);
+        }
+        return null;
+    }
+
+    @Override
+    public List<Attribute> getReturnAttributes() {
+        List<Attribute> attributes = new ArrayList<>();
+        attributes.add(new Attribute("isSuccess", BOOL));
+        return attributes;
+    }
+
+    @Override
+    public ProcessingMode getProcessingMode() {
+        return ProcessingMode.BATCH;
```

Comment:
Why this is BATCH?

Reply:
@grainier , This is the default value to be sent